### PR TITLE
4: Use secure banking BOM and spring BOM

### PR DESCRIPTION
**Pull request by BohoCode**
Also adds a read me and some more 3rd party dependencies
Issue: https://github.com/SecureBankingAcceleratorToolkit/SecureBankingAcceleratorToolkit/issues/4

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,9 +1,168 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
-  <modelVersion>4.0.0</modelVersion>
-  <groupId>com.forgerock.securebanking</groupId>
-  <artifactId>securebanking-parent</artifactId>
-  <version>1.0.0-SNAPSHOT</version>
-  <packaging>pom</packaging>
-  <name>securebanking-parent</name>
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <!--
```
Comment (jorgesanchezperez) [inline pom.xml:3]: The license Shouldn't in the top of the file?

**Review by jorgesanchezperez — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+</settings>
+
+```
+
+For further information see [here]([https://maven.apache.org/plugins/maven-compiler-plugin/examples/compile-using
+-different-jdk.html)
+
+## Distribution Management 
+
+Specifies the repositories into which artifacts should be built.
```
Comment (matt-wills) [inline README.md:47]: Perhaps add a TODO comment or remove this section?

**Review by jorgesanchezperez — DISMISSED** [review]

File: pom.xml (hunk 1 above)
Comment (BohoCode) [reply pom.xml]: Fixed (ish). It has to be after the xml decleration at the top of the file... which was missing!

File: pom.xml [hunk 3]
```diff
@@ -1,17 +1,169 @@
...
+                <version>2.10.8</version>
+            </dependency>
+
+
+            <!-- https://mvnrepository.com/artifact/org.projectlombok/lombok -->
+            <dependency>
+                <groupId>org.projectlombok</groupId>
+                <artifactId>lombok</artifactId>
+                <version>${lombok.version}</version>
+                <scope>provided</scope>
```
Comment (matt-wills) [inline pom.xml:126]: Out of interest, why is this `provided`?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  [...]
+  <activeProfiles>
+    <activeProfile>compiler</activeProfile>
+  </activeProfiles>
+</settings>
+
+```
+
+For further information see [here]([https://maven.apache.org/plugins/maven-compiler-plugin/examples/compile-using
+-different-jdk.html)
```
Comment (matt-wills) [inline README.md:43]: I clicked on this link but due to the newline it doesn't render correctly.

It seems odd that we need to set `JAVA_1_14_HOME ` in the maven compiler plugin for JDK 14, but not 11. Why is this?

File: README.md (hunk 2 above)
Comment (BohoCode) [reply README.md:47]: Have added more info in that section

File: pom.xml (hunk 3 above)
Comment (BohoCode) [reply pom.xml:126]: Hmm, copied from https://mvnrepository.com/artifact/org.projectlombok/lombok/1.18.16

I didn't spot it was provided! Makes no sense. Will remove the scope

File: README.md (hunk 4 above)
Comment (BohoCode) [reply README.md:43]: Due to a change in the latest maven-compiler-plugin. For the first time in years I can actually kick off the maven install lifecycle from in intellij and it works now. This is magic stuff :-)

Comment (BohoCode) [reply README.md]: Fixed bad formatting

**Review by matt-wills — APPROVED** [review]

**Pull request merged**